File: src/main/java/com/medoptica/medoptica/models/OrderRequest.java
package com.medoptica.medoptica.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderRequest {
    private String fullName;
    private String phone;
    private String region;
    private String city;
    private String branch;
    private String paymentMethod;
    private List<ProductCardDto> products;
}

File: src/main/front/medoptica/src/app/shared/models/order.request.ts
import {ProductCardDto} from './product-card-dto';

export class OrderRequest {
  fullName: string;
  phone: string;
  region: string;
  city: string;
  branch: string;
  paymentMethod: string;
  products: ProductCardDto[];

  constructor(fullName: string, phone: string, region: string, city: string, branch: string, paymentMethod: string, products: ProductCardDto[]) {
    this.fullName = fullName;
    this.phone = phone;
    this.region = region;
    this.city = city;
    this.branch = branch;
    this.paymentMethod = paymentMethod;
    this.products = products;
  }
}
